.travis.yml

```diff
@@ -1,89 +1,149 @@
+
+    - <<: *build-examples
+      name: "tvOS 12.3"
+      env: DESTINATION="$TVOS_DESTINATION_12" WORKSPACE="$TVOS_EXAMPLE_WORKSPACE" SDK="$TVOS_SDK" SCHEME="$EXAMPLE_SCHEME"
+
+    - <<: *build-examples
+      name: "tvOS 11.3"
+      env: DESTINATION="$TVOS_DESTINATION_11" WORKSPACE="$TVOS_EXAMPLE_WORKSPACE" SDK="$TVOS_SDK" SCHEME="$EXAMPLE_SCHEME"
+
+    - <<: *build-examples
+      name: "tvOS 10.3"
+      env: DESTINATION="$TVOS_DESTINATION_10" WORKSPACE="$TVOS_EXAMPLE_WORKSPACE" SDK="$TVOS_SDK" SCHEME="$EXAMPLE_SCHEME"
+
+    # UI Tests
+    - stage: tests
+      name: "UI Tests: iOS 12.2"
+      env: DESTINATION="$IOS_DESTINATION_12" WORKSPACE="$IOS_EXAMPLE_WORKSPACE" SDK="$IOS_SDK" SCHEME="$EXAMPLE_SCHEME"
+      script:
+        - set -o pipefail
+        - xcodebuild build test -workspace "$WORKSPACE" -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
+
+    # Unit Tests
+    - &unit-tests
+      stage: tests
+      name: "Tests: iOS 12.2"
+      env: DESTINATION="$IOS_DESTINATION_12" WORKSPACE="IGListKit.xcworkspace" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME"
+      script:
+        - set -o pipefail
+        - xcodebuild build build-for-testing -workspace "$WORKSPACE" -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ONLY_ACTIVE_ARCH=YES | bundle exec xcpretty -c;
+        - xcodebuild analyze test-without-building -workspace "$WORKSPACE" -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=YES GCC_GENERATE_TEST_COVERAGE_FILES=YES ONLY_ACTIVE_ARCH=YES | bundle exec xcpretty -c;
+      after_success:
+        - bundle exec slather
+
+    - <<: *unit-tests
+      name: "Tests: iOS 11.3"
+      env: DESTINATION="$IOS_DESTINATION_11" WORKSPACE="IGListKit.xcworkspace" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME"
+
+    # Analysis
+    - &static-analysis
+      stage: tests
+      name: "Static Analysis: iOS"
+      env: DESTINATION="$IOS_DESTINATION_12" WORKSPACE="IGListKit.xcworkspace" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME"
+      script: xcodebuild build analyze -workspace "$WORKSPACE" -scheme "$SCHEME" -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
+
+    - <<: *static-analysis
+      name: "Static Analysis: macOS"
+      env: DESTINATION="$MACOS_DESTINATION" WORKSPACE="IGListKit.xcworkspace" SDK="$TVOS_SDK" SCHEME="$MACOS_SCHEME"
+
+    - <<: *static-analysis
+      name: "Static Analysis: iOS"
+      env: DESTINATION="$TVOS_DESTINATION_12" WORKSPACE="IGListKit.xcworkspace" SDK="$TVOS_SDK" SCHEME="$TVOS_SCHEME"
+
+    # Post stage
+    - stage: post test
+      name: Danger
+      script: bundle exec danger --verbose
```

Comment:
we should remove this stage and add this to Unit Tests `after_success:`

Reply:
I created a separate stage because it wouldn't run if the tests failed and would only run once. If we attach it to the `after_success`, I'd have to make sure it only runs on one of the instances which feels a bit awkward.

Reply:
ah. fair enough. could we name this stage "Danger" then?

Reply:
Aye! Will do

Reply:
Done!

Reply:
👍